modules/reactor/src/meshgenerators/ControlDrumMeshGenerator.C

```diff
@@ -0,0 +1,509 @@
+      }
+      if (!MooseUtils::absoluteFuzzyEqual(current_azim_angle, custom_end_angle) &&
+          !MooseUtils::absoluteFuzzyEqual(next_azim_angle, custom_end_angle) &&
+          (custom_end_angle > current_azim_angle) && (custom_end_angle < next_azim_angle))
+      {
+        mooseWarning("pad_end_angle not contained within drum azimuthal discretization so "
+                     "additional azimuthal nodes are created to align mesh with this angle");
+        azimuthal_angles.push_back(custom_end_angle);
+      }
+    }
+    {
+      // Invoke AdvancedConcentricCircleGenerator to define drum mesh without background region
+      auto params = _app.getFactory().getValidParams("AdvancedConcentricCircleGenerator");
+
+      params.set<std::vector<Real>>("customized_azimuthal_angles") = azimuthal_angles;
+      params.set<std::vector<double>>("ring_radii") = {_drum_inner_radius, _drum_outer_radius};
+      params.set<std::vector<unsigned int>>("ring_intervals") = {drum_inner_intervals,
+                                                                 drum_intervals};
+      if (drum_inner_intervals > 1)
+      {
+        params.set<std::vector<unsigned short>>("ring_block_ids") = {
+            RGMB::CONTROL_DRUM_BLOCK_ID_INNER_TRI,
+            RGMB::CONTROL_DRUM_BLOCK_ID_INNER,
+            RGMB::CONTROL_DRUM_BLOCK_ID_PAD};
+        params.set<std::vector<SubdomainName>>("ring_block_names") = {
+            block_name_prefix + "_R0_TRI", block_name_prefix + "_R0", block_name_prefix + "_R1"};
+      }
+      else
+      {
+        params.set<std::vector<unsigned short>>("ring_block_ids") = {
+            RGMB::CONTROL_DRUM_BLOCK_ID_INNER, RGMB::CONTROL_DRUM_BLOCK_ID_PAD};
+        params.set<std::vector<SubdomainName>>("ring_block_names") = {block_name_prefix + "_R0",
+                                                                      block_name_prefix + "_R1"};
+      }
+      params.set<bool>("create_outward_interface_boundaries") = false;
+
+      addMeshSubgenerator("AdvancedConcentricCircleGenerator", name() + "_accg", params);
+    }
+    {
+      // Invoke FlexiblePatternGenerator to triangulate drum background region
+      auto params = _app.getFactory().getValidParams("FlexiblePatternGenerator");
+
+      params.set<std::vector<MeshGeneratorName>>("inputs") = {name() + "_accg"};
+      params.set<std::vector<libMesh::Point>>("extra_positions") = {libMesh::Point(0, 0, 0)};
+      params.set<std::vector<unsigned int>>("extra_positions_mg_indices") = {0};
+      params.set<bool>("use_auto_area_func") = true;
+      params.set<bool>("verify_holes") = false;
+      params.set<MooseEnum>("boundary_type") = (_geom_type == "Hex") ? "HEXAGON" : "CARTESIAN";
+      params.set<unsigned int>("boundary_sectors") =
+          getReactorParam<unsigned int>(RGMB::num_sectors_flexible_stitching);
+      params.set<Real>("boundary_size") = assembly_pitch;
+      params.set<boundary_id_type>("external_boundary_id") = 20000 + _assembly_type;
```

Comment:
do we have this 20000 in RGMB namespace ? it d be good

Reply:
Good catch, the RGMB namespace variable is used here instead